app/src/test/java/org/simple/clinic/editpatient/PatientEditScreenControllerTest.kt

```diff
@@ -782,4 +786,26 @@ class PatientEditScreenControllerTest {
     verify(screen, times(2)).showDatePatternInDateOfBirthLabel()
     verify(screen).hideDatePatternInDateOfBirthLabel()
   }
+
+  @Test
+  fun `date-of-birth and age fields should only be visible while one of them is empty`() {
+    uiEvents.onNext(PatientEditAgeTextChanged(""))
+    uiEvents.onNext(PatientEditDateOfBirthTextChanged(""))
+    verify(screen).setDateOfBirthAndAgeVisibility(DateOfBirthAndAgeVisibility.BOTH_VISIBLE)
+
+    uiEvents.onNext(PatientEditDateOfBirthTextChanged("1"))
+    verify(screen).setDateOfBirthAndAgeVisibility(DateOfBirthAndAgeVisibility.DATE_OF_BIRTH_VISIBLE)
+
+    uiEvents.onNext(PatientEditDateOfBirthTextChanged(""))
+    uiEvents.onNext(PatientEditAgeTextChanged("1"))
+    verify(screen).setDateOfBirthAndAgeVisibility(DateOfBirthAndAgeVisibility.AGE_VISIBLE)
+  }
+
+  @Test()
+  fun `when both date-of-birth and age fields have text then an assertion error should be thrown`() {
+    errorConsumer = { Truth.assertThat(it).isInstanceOf(AssertionError::class.java) }
```

Comment:
Hrm, this is also an interesting way of handling errors instead of using `ExpectedException`.

Reply:
Oh, I just used whatever was done in `PatientEntryScreenControllerTest`.